packages/node/src/BacktraceNodeRequestHandler.ts

```diff
@@ -0,0 +1,155 @@
+                            result += d.toString();
+                        });
+                        response.on('end', () => {
+                            switch (response.statusCode) {
+                                case 200: {
+                                    res(BacktraceReportSubmissionResult.Ok(this.parseServerResponse(result)));
+                                    break;
+                                }
+                                case 401:
+                                case 403: {
+                                    res(BacktraceReportSubmissionResult.OnInvalidToken());
+                                    break;
+                                }
+                                case 429: {
+                                    res(BacktraceReportSubmissionResult.OnLimitReached());
+                                    break;
+                                }
+                                default: {
+                                    res(BacktraceReportSubmissionResult.OnInternalServerError(result));
+                                    break;
+                                }
+                            }
+                        });
+                    },
+                );
+
+                request.on('error', (err: Error) => {
+                    if (ConnectionError.isConnectionError(err)) {
+                        return res(BacktraceReportSubmissionResult.OnNetworkingError(err.message));
+                    }
+                    return res(BacktraceReportSubmissionResult.OnInternalServerError(err.message));
+                });
+
+                if (typeof payload === 'string') {
+                    request.write(payload);
+                } else {
+                    payload.pipe(request);
+                }
+
+                request.end();
+            });
+        } catch (err) {
+            if (ConnectionError.isConnectionError(err)) {
+                return BacktraceReportSubmissionResult.OnNetworkingError(err.message);
+            }
+
+            const errorMessage = err instanceof Error ? err.message : (err as string);
+            return BacktraceReportSubmissionResult.OnUnknownError(errorMessage);
+        }
+    }
+
+    private getHttpClient(submissionUrl: URL) {
+        return submissionUrl.protocol === 'http' ? http : https;
+    }
+
+    private parseServerResponse<T>(response: string): T {
+        try {
+            return JSON.parse(response);
+        } catch {
+            return {} as T;
+        }
+    }
+    private createFormData(json: string, attachments?: BacktraceAttachment[]) {
+        const formData = new FormData();
+        formData.append(this.UPLOAD_FILE_NAME, json, `${this.UPLOAD_FILE_NAME}.json`);
+
+        if (!attachments || attachments.length === 0) {
+            return formData;
+        }
+
+        for (const attachment of attachments) {
+            if (typeof attachment === 'string') {
+                if (!fs.existsSync(attachment)) {
```

Comment:
Is there a reason why we're not using the promise/callback version of this?

Reply:
I still believe we're still figuring out the minimum supported version, this is why I preferred to use the API that definitely works everythwere.

Reply:
Fair enough. But you could still promisify the callback version.